wiki/development/coding-guidelines.md

```diff
@@ -80,127 +238,233 @@ class Two {foo = 2;};
+Functions and code blocks that specific a return a value must have a meaningfull return value.
 
-[CBA script_macros_common.hpp](https://gist.github.com/commy2/9ed6cc73fbe6a2b3f4e1)
+### 6.5. Private Variables
+All private variables shall be declared in a private array.
+```
+private [“_myVariable”];
+```
 
-* `QUOTE()` is utilized within configuration files for bypassing the quote issues in configuration macros. So, all code segments inside a given config should utilize wrapping in the QUOTE() macro instead of direct strings. This allows us to use our macros inside the string segments, such as `QUOTE(_this call FUNC(balls))`
+### 6.6. Lines of Code
+Any one function shall contain no more than 250 lines of code, excluding the function header and any includes.
 
-#### 4.2.1 setVariable, getVariable family macros
+### 6.7. Variable declarations
+Declarations should be at the smallest feasible scope.
 
-| Macro  | Expands to |
-| -------|---------|
-|`GETVAR(player,MyVarName,false)` | player getVariable ["MyVarName", false]|
-|`GETMVAR(MyVarName,objNull)` | missionNamespace getVariable ["MyVarName", objNull]|
-|`GETUVAR(MyVarName,displayNull)` | uiNamespace getVariable ["MyVarName", displayNull]|
-|`SETVAR(player,MyVarName,127)` |  player setVariable ["MyVarName", 127]  SETPVAR(player,MyVarName,127) player setVariable ["MyVarName", 127, true] |
-|`SETMVAR(MyVarName,player)` | missionNamespace setVariable ["MyVarName", player] |
-|`SETUVAR(MyVarName,_control)` | uiNamespace setVariable ["MyVarName", _control] |
+### 6.8. Variable initialization
+Private variables will not be introduced until they can be initialized with meaningful values.
 
-#### 4.2.2 STRING family macros
+Good:
+```
+_myVariable = 0;
+{
+    _x params ["_value", "_amount"];
+    if (_value > 0) then {
+        _myVariable = _myVariable + _amount;
+    };
+} forEach _array;
+```
+Bad:
+```
+_myvariable = 0;
+if (_condition) then {
+    _myVariable = 1;
+} else {
+    _myvariable = 2;
+};
+```
+Good:
+```
+_myvariable = [1, 2] select _condition;
+```
 
-Note that you need the strings in module stringtable.xml in the correct format
-`STR_ACE_<module>_<string>`</br>
-Example:</br>
-`STR_Balls_Banana`</br>
+### 6.9. Initialization expression in for loops
+The initialization expression in a for loop will perform no actions other than to initialize the value of a single for loop parameter.
 
-Script strings:
+### 6.10. Increment expression in for loops
+The increment expression in a for loop will perform no action other than to change a single loop parameter to the next value for the loop.
 
-| Macro  |  Expands to |
-| -------|---------|
-| `LSTRING(banana)` |  "STR_ACE_balls_banana"|
-| `ELSTRING(balls,banana)` | "STR_ACE_balls_banana"|
+### 6.11. GetVariable
+When using getvariable, there should either be a default value  given in the statement or the return value should be checked for correct data type as well as return value. A default value may not be given after a nil check.
```

Comment:
When using `getVariable`, there should either be a default value given in the statement or the return value should be checked for correct data type as well as return value. A default value may not be given after a nil check.